res/controllers/Pioneer-DDJ-SB2-scripts.js

```diff
@@ -0,0 +1,1187 @@
+    } else {
+        engine.connectControl('[Channel1]', 'VuMeter', 'PioneerDDJSB2.VuMeterLeds', isUnbinding);
+        engine.connectControl('[Channel2]', 'VuMeter', 'PioneerDDJSB2.VuMeterLeds', isUnbinding);
+        engine.connectControl('[Channel3]', 'VuMeter', 'PioneerDDJSB2.VuMeterLeds', isUnbinding);
+        engine.connectControl('[Channel4]', 'VuMeter', 'PioneerDDJSB2.VuMeterLeds', isUnbinding);
+    }
+};
+
+PioneerDDJSB2.bindAllControlConnections = function(isUnbinding) {
+    var samplerIndex,
+        fxUnitIndex,
+        channelIndex;
+
+    for (samplerIndex = 1; samplerIndex <= 4; samplerIndex++) {
+        PioneerDDJSB2.bindSamplerControlConnections('[Sampler' + samplerIndex + ']', isUnbinding);
+    }
+
+    for (fxUnitIndex = 1; fxUnitIndex <= 2; fxUnitIndex++) {
+        engine.connectControl('[EffectRack1_EffectUnit' + fxUnitIndex + ']', 'group_[Headphone]_enable', 'PioneerDDJSB2.fxLeds', isUnbinding);
+    }
+
+    for (channelIndex = 1; channelIndex <= 2; channelIndex++) {
+        PioneerDDJSB2.bindDeckControlConnections('[Channel' + channelIndex + ']', isUnbinding);
+    }
+};
+
+PioneerDDJSB2.setDeckSoftTakeover = function(channel, isUnbinding) {
+    engine.softTakeover(channel, "volume", !isUnbinding);
+    engine.softTakeover(channel, "rate", !isUnbinding);
+    engine.softTakeover(channel, "pregain", !isUnbinding);
+    engine.softTakeover(channel, "filterHigh", !isUnbinding);
+    engine.softTakeover(channel, "filterMid", !isUnbinding);
+    engine.softTakeover(channel, "filterLow", !isUnbinding);
+    engine.softTakeover("[QuickEffectRack1_" + channel + "]", "super1", !isUnbinding);
+};
+
+PioneerDDJSB2.setAllSoftTakeover = function(isUnbinding) {
+    var channelIndex;
+    for (channelIndex = 1; channelIndex <= 4; channelIndex++) {
+        PioneerDDJSB2.setDeckSoftTakeover('[Channel' + channelIndex + ']', isUnbinding);
+    }
+};
+
+
+///////////////////////////////////////////////////////////////
+//                       DECK SWITCHING                      //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSB2.deckSwitchTable = {
+    '[Channel1]': '[Channel1]',
+    '[Channel2]': '[Channel2]',
+    '[Channel3]': '[Channel3]',
+    '[Channel4]': '[Channel4]'
+
+};
+
+PioneerDDJSB2.deckShiftSwitchTable = {
+    '[Channel1]': '[Channel3]',
+    '[Channel2]': '[Channel4]',
+    '[Channel3]': '[Channel1]',
+    '[Channel4]': '[Channel2]'
+};
+
+PioneerDDJSB2.initDeck = function(group) {
+    PioneerDDJSB2.bindDeckControlConnections(group, false);
+    PioneerDDJSB2.nonPadLedControl(group, PioneerDDJSB2.nonPadLeds.shiftKeyLock, PioneerDDJSB2.channelGroups[group] > 1);
+    PioneerDDJSB2.triggerVinylLed(PioneerDDJSB2.channelGroups[group]);
+};
+
+
+///////////////////////////////////////////////////////////////
+//            HIGH RESOLUTION MIDI INPUT HANDLERS            //
+///////////////////////////////////////////////////////////////
+
```

Comment:
Tested, but this is a greater work. Reason: The code based on old DDj-SB, a 2 deck controller and the 4 deck mode was/is realised "virtual". So this is often used in code.
Eleminate this is on my todo.